From cb40b645f2b7f3dfe6d24dbb88179a42dccb4176 Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Mon, 21 Mar 2016 08:14:03 +0100
Subject: [PATCH] S Core: Add boot for alternativ kernel dtb

Add a possibilty to boot an alternate kernel and dtb. There is only a
possibility to boot an main kernel and dtb or the alternate kernel
and dtb. The process can be controlled via the kernel_alt variable from
the u-boot environment. If the variable is 0 (default) the main kernel
and dtb will be booted. If the variable is 1 boot the alternative ones.

Ticket #1046
---
 include/configs/mx6sxscore.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/configs/mx6sxscore.h b/include/configs/mx6sxscore.h
index 5fda3b3..84dadde 100644
--- a/include/configs/mx6sxscore.h
+++ b/include/configs/mx6sxscore.h
@@ -169,15 +169,16 @@
     "image=uImage\0" \
     "fdt_file=imx6sx-score.dtb\0" \
     "add_args=\0" \
-    "mtdparts=gpmi-nand:512k(ubootenv),8m(kernel),2m(dtb),8m(kernel_backup),2m(dinfo),-(rootfs)\0" \
+    "mtdparts=gpmi-nand:512k(ubootenv),8m(kernel),2m(dtb),8m(kernel_backup),2m(dtb_backup),-(rootfs)\0" \
     "update_nor=fatload mmc 1:1 ${loadaddr} u-boot.imx && sf probe && sf erase 0x0 0x60000 && sf write ${loadaddr} 0x400 0x60000\0"
 
 #define CONFIG_BOOT_KERNEL_ROOTFS_NAND \
     "getrootname=if test $root_alt -eq 0; then setenv rootname rootfs; else; setenv rootname rootfs_alt; fi;\0" \
     "setrootnand=setenv rootnand ubi0:${rootname} rootfstype=ubifs\0" \
     "root_alt=0\0" \
-    "nandkernel=nand read ${loadaddr} 0x80000 0x800000\0" \
-    "nandfdt=nand read ${fdt_addr} 0x880000 0x100000\0" \
+    "kernel_alt=0\0" \
+    "nandkernel=if test $kernel_alt -eq 0; then nand read ${loadaddr} 0x80000 0x800000; else; nand read ${loadaddr} 0xa80000 0x800000; fi;\0" \
+    "nandfdt=if test $kernel_alt -eq 0 ; then nand read ${fdt_addr} 0x880000 0x200000; else; nand read ${fdt_addr} 0x1280000 0x200000; fi;\0" \
     "nandargs=setenv bootargs " \
         " console=${console},${baudrate} panic=2 uart_from_osc ubi.mtd=5 root=${rootnand} mtdparts=${mtdparts} ${add_args}\0" \
     "bootnand=run getrootname && " \
-- 
1.9.1

